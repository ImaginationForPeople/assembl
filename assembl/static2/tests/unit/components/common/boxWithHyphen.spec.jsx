import React from 'react';
import renderer from 'react-test-renderer';

import BoxWithHyphen from '../../../../js/app/components/common/boxWithHyphen';
import '../../../helpers/setupTranslations';

describe('BoxWithHyphen component', () => {
  it('should render a box with hyphen with its data', () => {
    const props = {
      additionalContainerClassNames: 'foobar',
      body: 'Lorem ipsum dolor sit amet',
      date: '2017-01-01T21:05:25.085948Z',
      href: 'http://www.example.com/foobar',
      hyphenStyle: {
        borderColorTop: 'red'
      },
      subject: 'Foobar',
      title: 'Synthesis'
    };
    const component = renderer.create(<BoxWithHyphen {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not render a link that wraps the box if href is empty', () => {
    const props = {
      additionalContainerClassNames: 'foobar',
      body: 'Lorem ipsum dolor sit amet',
      date: '2017-01-01T21:05:25.085948Z',
      href: '',
      hyphenStyle: {
        borderColorTop: 'red'
      },
      subject: 'Foobar',
      title: 'Synthesis'
    };
    const component = renderer.create(<BoxWithHyphen {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});