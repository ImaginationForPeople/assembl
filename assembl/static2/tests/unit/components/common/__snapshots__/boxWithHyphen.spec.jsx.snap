// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxWithHyphen component should not render a link that wraps the box if href is empty 1`] = `
<div
  className="foobar box-with-hyphen"
>
  <div
    className="insert-box"
  >
    <h3
      className="dark-title-4"
    >
      <div>
        Synthesis
      </div>
      <div
        className="ellipsis margin-xs"
      >
        Foobar
      </div>
    </h3>
    <div
      className="box-hyphen"
      style={
        Object {
          "borderColorTop": "red",
        }
      }
    >
       
    </div>
    <div
      className="date"
    >
      <span
        className={undefined}
        style={undefined}
      >
        2017-01-01
      </span>
    </div>
    <div
      className="insert-content margin-s"
    >
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "Lorem ipsum dolor sit amet",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`BoxWithHyphen component should render a box with hyphen with its data 1`] = `
<div
  className="foobar box-with-hyphen"
>
  <a
    href="http://www.example.com/foobar"
  >
    <div
      className="insert-box"
    >
      <h3
        className="dark-title-4"
      >
        <div>
          Synthesis
        </div>
        <div
          className="ellipsis margin-xs"
        >
          Foobar
        </div>
      </h3>
      <div
        className="box-hyphen"
        style={
          Object {
            "borderColorTop": "red",
          }
        }
      >
         
      </div>
      <div
        className="date"
      >
        <span
          className={undefined}
          style={undefined}
        >
          2017-01-01
        </span>
      </div>
      <div
        className="insert-content margin-s"
      >
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Lorem ipsum dolor sit amet",
            }
          }
        />
      </div>
    </div>
  </a>
</div>
`;
